pkg/virt-controller/ipamclaims/ipamclaims_test.go

```diff
@@ -0,0 +1,337 @@
+			libvmi.WithInterface(virtv1.Interface{Name: "absent", State: virtv1.InterfaceStateAbsent}),
+		)
+		vmi.UID = vmiUID
+	})
+
+	Context("With allowPersistentIPs enabled in the NADs", func() {
+		var ipamClaimsManager *ipamclaims.IPAMClaimsManager
+
+		BeforeEach(func() {
+			persistentIPs := map[string]struct{}{redNetworkNadName: {}, blueNetworkNadName: {}}
+			Expect(createNADs(networkClient, vmi.Namespace, vmi.Spec.Networks, persistentIPs)).To(Succeed())
+
+			ipamClaimsManager = ipamclaims.NewIPAMClaimsManager(networkClient, ipamClaimsClient)
+		})
+
+		It("should create the expected IPAMClaims", func() {
+			ownerRef := &v1.OwnerReference{
+				APIVersion:         "kubevirt.io/v1",
+				Kind:               "VirtualMachine",
+				Name:               vmi.Name,
+				UID:                vmUID,
+				Controller:         pointer.P(true),
+				BlockOwnerDeletion: pointer.P(true),
+			}
+			Expect(ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)).To(Succeed())
+
+			ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(vmi.Namespace).List(
+				context.Background(),
+				v1.ListOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaimsList.Items).To(HaveLen(2))
+			assertIPAMClaim(ipamClaimsList.Items[0], vmi.Namespace, vmi.Name, blueNetworkLogicalName, "pod16477688c0e", "VirtualMachine")
+			assertIPAMClaim(ipamClaimsList.Items[1], vmi.Namespace, vmi.Name, redNetworkLogicalName, "podb1f51a511f1", "VirtualMachine")
+		})
+
+		Context("When IPAMClaims already exist", func() {
+			var ownerRef *v1.OwnerReference
+
+			BeforeEach(func() {
+				ownerRef = &v1.OwnerReference{
+					APIVersion:         "kubevirt.io/v1",
+					Kind:               "VirtualMachine",
+					Name:               vmi.Name,
+					UID:                vmUID,
+					Controller:         pointer.P(true),
+					BlockOwnerDeletion: pointer.P(true),
+				}
+				Expect(ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)).To(Succeed())
+
+				ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(vmi.Namespace).List(
+					context.Background(),
+					v1.ListOptions{},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ipamClaimsList.Items).To(HaveLen(2))
+			})
+
+			It("with the expected owner UID, should not fail re-creation/validation attempt", func() {
+				Expect(ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)).To(Succeed())
+
+				ipamClaimsList, err := ipamClaimsClient.K8sV1alpha1().IPAMClaims(vmi.Namespace).List(
+					context.Background(),
+					v1.ListOptions{},
+				)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ipamClaimsList.Items).To(HaveLen(2))
+			})
+
+			It("with a different owner UID, should fail fail re-creation/validation attempt", func() {
+				ownerRef.UID = "differentUID"
+				err := ipamClaimsManager.CreateNewPodIPAMClaims(vmi, ownerRef)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("wrong IPAMClaim with the same name still exists"))
```

Comment:
nit: I think you can mix these 2 up. 

```suggestion
Expect(err).To(MatchError(ContainSubstring("wrong IPAMClaim with the same name still exists")))
```